query-engine/core/src/interpreter/query_interpreters/nested_read.rs

```diff
@@ -221,7 +212,7 @@ pub async fn one2m(
         let mut additional_records = vec![];
 
         for mut record in scalars.records.iter_mut() {
-            let child_link: RecordProjection = record.projection(&scalars.field_names, &child_link_id)?;
+            let child_link: SelectionResult = record.extract_selection_result(&scalars.field_names, &child_link_id)?;
```

Comment:
This makes so much more sense to me. I don't know if `SelectionResult` is the ultimate naming, but it leads to much simpler namings. Like said once, I'd still love to rename `Record` to something even more obvious, with a prefix or suffix that makes it obvious it contains values.